crates/extensions/tedge_timer_ext/src/actor.rs

```diff
@@ -195,6 +195,8 @@ impl Actor for TimerActor {
 
                         if let Some(timer_request) = maybe_message {
                             self.push(timer_request);
+                        } else {
```

Comment:
I didn't fully get how this is working. I was expecting this actor to use the `try_recv()` function and break out of this select loop on `Err(RuntimeRequest::Shutdown)` . But, it appears that somehow the message_box itself is getting closed on shutdown so that we can achieve the same using `None` response. Could you point me to the piece of code that does this?

Reply:
This is the behavior of [`CombinedReceiver::recv()`](https://github.com/thin-edge/thin-edge.io/blob/main/crates/core/tedge_actors/src/message_boxes.rs#L308) and is inherited by the `LoggingReceiver`, the `SimpleMessagebox` the `ServiceMessageBox`.

I will give a try to an explicitly call to [`CombinedReceiver::try_recv()`](https://github.com/thin-edge/thin-edge.io/blob/main/crates/core/tedge_actors/src/message_boxes.rs#L286).